.github/workflows/pr.yaml

```diff
@@ -54,11 +54,65 @@ jobs:
         with:
           go-version-file: 'go.mod'
 
+      - name: Check for outdated bundle
+        id: bundle-diff-checker
+        # Lot of debate (https://github.com/janus-idp/operator/pull/195) whether this should be a warning or an error.
+        # For now, this is will be warning + a comment on the PR if manifests are outdated. This way, PR authors/maintainers can be aware of that fact.
+        continue-on-error: true
+        run: |
+          make bundle
+          git status --porcelain
+          # Since operator-sdk 1.26.0, `make bundle` changes the `createdAt` field from the bundle every time we run it.
+          # The `git diff` below checks if only the createdAt field has changed. If is the only change, it is ignored.
+          # Inspired from https://github.com/operator-framework/operator-sdk/issues/6285#issuecomment-1415350333
+          git diff --quiet -I'^    createdAt: ' bundle || ( echo "::group::WARNINGS" && \
+            git --no-pager diff | tee bundle.pr-${{ github.event.number }}.patch && \
+            echo "::warning:: Files changed in bundle generation. Please regenerate the bundle with 'make bundle' and push the changes. For your convenience, the diff is attached as a job artifact, so you can easily download and Git-apply it right away instead. You might also need to manually update the CSV in '.rhdh/bundle/manifests/rhdh-operator.csv.yaml' file accordingly." && \
+            echo "::endgroup::" && \
+            exit 1 )
+
+      - name: Save bundle diff as patch
+        id: bundle-diff-patch-artifact-upload
+        uses: actions/upload-artifact@v4
+        if: ${{ !cancelled() && steps.bundle-diff-checker.outcome == 'failure' }}
+        with:
+          name: bundle-diff-patch
+          path: bundle.pr-${{ github.event.number }}.patch
```

Comment:
if you can generate a patch, why not just push a commit with this change?

Reply:
Yeah, initially, I was thinking of making the GitHub job just commit the diff, so that the PR author does not even need to think about it (Gennady's concern, I think).
But then I thought that the CSV in .rhdh might also need to be changed, and I am not sure we can handle that automatically.
Now with hindsight, I think it can indeed make sense to:
- automatically commit any diff with this change, which should sort out @gazarenkov's main concern, I think
- comment on the PR (or update the PR template) that the CSV in .rhdh should also be checked

WDYT? @gazarenkov @jianrongzhang89 @nickboldt

Reply:
Okay, I've gone ahead and implemented as proposed above:
- bundle manifests changes will be automatically pushed to the PR branch. Due to permission restrictions when using the built-in `GITHUB_TOKEN` token, this had to be done in a separate workflow triggered on `pull_request_target` events (+ same safeguard permissions we are already using to prevent PWN requests from external PRs)
- comment on the PR + update to the PR template that the CSV in `.rhdh` should also be checked

PTAL

/retitle Auto-generate bundle manifests changes to PR branch if needed